src/data/index.ts

```diff
@@ -0,0 +1,69 @@
+import sqlite3, { Database } from 'better-sqlite3'
+import { existsSync, mkdirSync } from "fs"
+import initBotData from "./initializers/botData"
+import { Databases, Pragma } from './types'
+
+const rootDirectory = process.cwd()
+const dataDirectory = rootDirectory + "/src/data/database/"
+if (!existsSync(dataDirectory)) {
+    try {
+        mkdirSync(dataDirectory)
+    } catch (error) {
+        console.log(error)
+    }
+}
+
+const loadedDatabases: {
+    [key in Databases]?: Database
+} = {}
+
+const dbPragmas = {
+    [Pragma.DEFAULT]: "journal_mode = WAL"
+}
+
+const databaseMetadata = {
+    [Databases.BOT_DATA]: {
+        path: 'bot_data.db',
+        pragma: Pragma.DEFAULT,
+        init: initBotData
+    }
+}
+
+export default function getDatabase(
+    database: Databases
+): Database {
+    const isLoaded = loadedDatabases[database]
+    if (isLoaded) {
+        return isLoaded
+    }
+
+    // load the database
+    const metadata = databaseMetadata[database]
+
+    const fullPath = dataDirectory + metadata.path
+
+    // see if the db already exists
+    const dbExists = existsSync(fullPath)
```

Comment:
Is there an `existsAsync` ? I'd like to use multithreading wherever possible

Reply:
The node.js file system library does provide asynchronous functions, but I decided to use the synchronous version here since  the getDatabase function needs to be synchronous.

Reply:
The node.js file system library does provide asynchronous functions, but I decided to use the synchronous version here since the getDatabase function needs to be synchronous.